### ISSUE TITLE
Segfaults when using invalid arguments

### ISSUE THREAD

**Issue by reox** [tataratat/splinepy#185]
During clueless and clumsy playing around with splinepy (0.0.26 from pypi), I triggerd several segfaults, when passing invalid arguments to functions.
I found some more, but have not saved the programs that crashed...
I used larry to run those programs, with anaconda 2021.11 (py3.9).

Invalid parametric coordinate when calling jacobian:
```
import numpy as np
import splinepy

bspline1 = splinepy.BSpline(
        degrees=[3],
        knot_vectors=[[0, 0, 0, 0, 1, 1, 1, 1]],
        control_points=np.array([[0.0, 0.0], [0.2, 3.0], [1.2, 2.0], [2.0, 2.0]]),
)

print(bspline1.jacobian([[1.0]]))  # OK
print(bspline1.sample(10))  # OK
print(bspline1.jacobian([[1.1]]))  # Segfault
```

again, but with unclamped knot vector, here also `sample` crashes:
```
import numpy as np
import splinepy

bspline1 = splinepy.BSpline(
        degrees=[3],
        # Create an unclamped, uniform knot vector
        knot_vectors=[[0, 1, 2, 3, 4, 5, 6, 7]],
        control_points=np.array([[0.0, 0.0], [0.2, 3.0], [1.2, 2.0], [2.0, 2.0]]),
)

print(bspline1.knot_vector_bounds)

print(bspline1.jacobian([[3.5]]))  # OK
print(bspline1.jacobian([[4.0]]))  # segfault
print(bspline1.sample(10))  # Segfault 
```

There is also a segfault when calling `bspline1.show()` either in a jupyter notebook or in an environment that does not have a physical GPU but only MESA. Not sure if you can detect that and simply write an error message or if that has to be fixed in vedo?

**Comment by jzwar** [comment]
Thank you for documenting these issues! We will definitely need more meaningful Exceptions in these cases.

**Comment by jzwar** [comment]
I split up the problem into two sections:
- [ ] Checks in constructors
- [ ] Checks in requests (jacobian, derivatives, evaluations,...)

The second might be better located on the python side

**Comment by jzwar** [comment]
Although the code will still fail, at least, it will now throw comprehensive error messages. It will be available in pip installable version `0.0.28`
